eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FullText.php

```diff
@@ -223,5 +240,43 @@ public function handle( CriteriaConverter $converter, SelectQuery $query, Criter
             $subSelect
         );
     }
+
+    /**
+     * Returns an exact content object count threshold to ignore common terms on.
+     *
+     * Common terms will be skipped if used in more then a given percentage of the total amount of content
+     * objects in the database. Percentage is defined by stopWordThresholdFactor configuration.
+     *
+     * Example: If stopWordThresholdFactor is 0.66 (66%), and a term like "the" exists in more then 66% of the content, it
+     *          will ignore the phrase as it is assumed to not add any value ot the search.
+     *
+     * Caches the result for the instance used as we don't need this to be super accurate as it is based on percentage,
+     * set by stopWordThresholdFactor.
+     *
+     * @return int
+     */
+    protected function getStopWordThresholdValue()
+    {
+        if ( $this->stopWordThresholdValue !== null )
+            return $this->stopWordThresholdValue;
+
+        // Cached value does not exists, do a simple count query on ezcontentobject table
+        $query = $this->dbHandler->createSelectQuery();
+        $query
+            ->select(
+                $query->alias( $query->expr->count( '*' ), 'count' )
+            )
+            ->from( $this->dbHandler->quoteTable( "ezcontentobject" ) );
+
+        $statement = $query->prepare();
+        $statement->execute();
+
+        // Makes sure factor is between 0 and 1
```

Comment:
ctor sounds nice. It should end up in the semantic config validation as
well, ideally (but it's a bonus).